model/histogram/histogram_test.go

```diff
@@ -410,3 +410,259 @@ func TestHistogramToFloat(t *testing.T) {
+				PositiveBuckets: []int64{0, 0, 1, -1, 0, 3, -2, 42, 3},
+			},
+			0,
+			&Histogram{
+				PositiveSpans:   []Span{{-2, 1}, {7, 4}},
+				PositiveBuckets: []int64{1, 2, -2, 42, 3},
+			},
+		},
+		{
+			"cut empty buckets from the middle of a span, avoiding some due to maxEmptyBuckets",
+			&Histogram{
+				PositiveSpans:   []Span{{-4, 6}, {3, 3}},
+				PositiveBuckets: []int64{0, 0, 1, -1, 0, 3, -2, 42, 3},
+				NegativeSpans:   []Span{{0, 2}, {3, 5}},
+				NegativeBuckets: []int64{5, 3, -4, -2, -2, 3, 4},
+			},
+			1,
+			&Histogram{
+				PositiveSpans:   []Span{{-2, 1}, {2, 1}, {3, 3}},
+				PositiveBuckets: []int64{1, 2, -2, 42, 3},
+				NegativeSpans:   []Span{{0, 2}, {3, 5}},
+				NegativeBuckets: []int64{5, 3, -4, -2, -2, 3, 4},
+			},
+		},
+		{
+			"avoiding all cutting of empty buckets from the middle of a chunk due to maxEmptyBuckets",
+			&Histogram{
+				PositiveSpans:   []Span{{-4, 6}, {3, 3}},
+				PositiveBuckets: []int64{0, 0, 1, -1, 0, 3, -2, 42, 3},
+				NegativeSpans:   []Span{{0, 2}, {3, 5}},
+				NegativeBuckets: []int64{5, 3, -4, -2, -2, 3, 4},
+			},
+			2,
+			&Histogram{
+				PositiveSpans:   []Span{{-2, 4}, {3, 3}},
+				PositiveBuckets: []int64{1, -1, 0, 3, -2, 42, 3},
+				NegativeSpans:   []Span{{0, 2}, {3, 5}},
+				NegativeBuckets: []int64{5, 3, -4, -2, -2, 3, 4},
+			},
+		},
+		{
+			"everything merged into one span due to maxEmptyBuckets",
+			&Histogram{
+				PositiveSpans:   []Span{{-4, 6}, {3, 3}},
+				PositiveBuckets: []int64{0, 0, 1, -1, 0, 3, -2, 42, 3},
+				NegativeSpans:   []Span{{0, 2}, {3, 5}},
+				NegativeBuckets: []int64{5, 3, -4, -2, -2, 3, 4},
+			},
+			3,
+			&Histogram{
+				PositiveSpans:   []Span{{-2, 10}},
+				PositiveBuckets: []int64{1, -1, 0, 3, -3, 0, 0, 1, 42, 3},
+				NegativeSpans:   []Span{{0, 10}},
+				NegativeBuckets: []int64{5, 3, -8, 0, 0, 4, -2, -2, 3, 4},
+			},
+		},
+		{
+			"only empty buckets and maxEmptyBuckets greater zero",
+			&Histogram{
+				PositiveSpans:   []Span{{-4, 6}, {3, 3}},
+				PositiveBuckets: []int64{0, 0, 0, 0, 0, 0, 0, 0, 0},
+				NegativeSpans:   []Span{{0, 7}},
+				NegativeBuckets: []int64{0, 0, 0, 0, 0, 0, 0},
+			},
+			3,
+			&Histogram{
+				PositiveSpans:   []Span{},
+				PositiveBuckets: []int64{},
+				NegativeSpans:   []Span{},
+				NegativeBuckets: []int64{},
+			},
+		},
+		{
+			"multiple spans of only empty buckets",
+			&Histogram{
+				PositiveSpans:   []Span{{-10, 2}, {2, 1}, {3, 3}},
+				PositiveBuckets: []int64{0, 0, 0, 0, 2, 3},
+				NegativeSpans:   []Span{{-10, 2}, {2, 1}, {3, 3}},
+				NegativeBuckets: []int64{2, 3, -5, 0, 0, 0},
+			},
+			0,
+			&Histogram{
+				PositiveSpans:   []Span{{-1, 2}},
+				PositiveBuckets: []int64{2, 3},
+				NegativeSpans:   []Span{{-10, 2}},
+				NegativeBuckets: []int64{2, 3},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			require.Equal(t, c.expected, c.in.Compact(c.maxEmptyBuckets))
+			// Has it also happened in-place?
```

Comment:
I feel it is easier to read test when we clearly state what needs to happen (else I have to read it as question and try to answer myself by looking at the code)
```suggestion
			// Compact also happens in-place.
```

Reply:
Sure. I also changed in in float_histogram_test.go.